### PR TITLE
Integration update heroku

### PR DESCRIPTION
**Pull request by vmatsiiako**
# Description 📣

## Type ✨

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Documentation

# Tests 🛠️

```sh
# Here's some code block to paste some code snippets
```

---

- [x] I have read the [contributing guide](https://infisical.com/docs/contributing/getting-started/overview), agreed and acknowledged the [code of conduct](https://infisical.com/docs/contributing/getting-started/code-of-conduct). 📝

### CODE REVIEW

File: docs/integrations/cloud/heroku.mdx [hunk 1]
```diff
@@ -30,6 +30,17 @@ description: "How to sync secrets from Infisical to Heroku"
...
+        - Secrets Path: The path in the current Infisical project from which you want to sync secrets from such as `/` (for secrets that do not reside in a folder) or `/foo/bar` (for secrets nested in a folder, in this case a folder called `bar` in another folder called `foo`).
+        - Heroku App: The application in Heroku that you want to sync secrets to.
+        - Initial Sync Behavior (default is **Import - Prefer values from Infisical**): The behavior of the first sync operation triggered after creating the integration.
+          - **No Import - Overwrite all values in Heroku**: Sync secrets and overwrite any existing secrets in Heroku.
+          - **Import - Prefer values from Infisical**: Import secrets from Heroku to Infisical; if a secret with the same name already exists in Infisical, do nothing. Afterwards, sync secrets to Heroku.
+          - **Import - Prefer values from Heroku**: Import secrets from Heroku to Infisical; if a secret with the same name already exists in Infisical, replace its value with the one from Heroku. Afterwards, sync secrets to Heroku.
```
Comment (akhilmhdh) [inline docs/integrations/cloud/heroku.mdx:42]: I think this should be a different value right. Both `import` and `import`

File: frontend/src/components/v2/RadioGroup/RadioGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
```
Comment (akhilmhdh) [inline frontend/src/components/v2/RadioGroup/RadioGroup.tsx:1]: Kindly avoid keeping non reusable components from components/v2. It should only contains reusable ui components. 

Ideally i would only keep a generic `RadioGroup` and `RadioGroupItem` as two separate components inside this file and then use it inside the heroku page.

I don't think this component will ever be used anywhere like this as its clearly containing heroku exclusive data

File: frontend/src/pages/integrations/heroku/create.tsx [hunk 3]
```diff
@@ -17,130 +34,376 @@ import {
...
-        setTargetApp("none");
+        setValue("targetApp", "none");
       }
     }
   }, [integrationAuthApps]);
 
-  const handleButtonClick = async () => {
-    try {
-      setIsLoading(true);
+  // const handleButtonClick = async () => {
```
Comment (akhilmhdh) [inline frontend/src/pages/integrations/heroku/create.tsx:144]: Is this needed anymore. I see lot of comments here

File: frontend/src/pages/integrations/heroku/create.tsx [hunk 4]
```diff
@@ -17,130 +34,376 @@ import {
...
+  useGetWorkspaceById 
+} from "../../../hooks/api/workspace";
+
+const initialSyncBehaviors = [
+  { label: "No Import - Overwrite all values in Heroku", value: IntegrationSyncBehavior.OVERWRITE_TARGET },
+  { label: "Import - Prefer values from Heroku", value: IntegrationSyncBehavior.PREFER_TARGET },
+  { label: "Import - Prefer values from Infisical", value: IntegrationSyncBehavior.PREFER_SOURCE }
+];
+
+const schema = yup.object({
```
Comment (akhilmhdh) [inline frontend/src/pages/integrations/heroku/create.tsx:48]: prefer zod

File: frontend/src/components/v2/RadioGroup/RadioGroup.tsx (hunk 2 above)
Comment (dangtony98) [reply frontend/src/components/v2/RadioGroup/RadioGroup.tsx:1]: Agree with this one; this should indeed contain generic components only.

Anyways, `RadioGroup` will not be used in this iteration of the Heroku integration update.

File: frontend/src/pages/integrations/heroku/create.tsx (hunk 3 above)
Comment (dangtony98) [reply frontend/src/pages/integrations/heroku/create.tsx:144]: We're still deciding on whether to go ahead with the fuller integration initiative to sync to Heroku pipelines, so just commenting out the logic; may come back to it in the future.

**Pull request merged**